File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    small = 0
    big = len(people) - 1
    while small <= big:
        if limit < people[big] + people[small]:
            big-=1
            answer+=1
        else:
            big-=1
            small+=1
            answer+=1
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
import java.io.*;
import java.util.stream.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        
        int min = 0;
        int max = people.length - 1;
        while(min <= max){
            if(people[min] + people[max] > limit ){
                max --;
            }else{
                min++;
                max--;
            }
            answer++;
        }
        return answer;
    }
}